beacon_node/network/src/sync/manager.rs

```diff
@@ -800,17 +874,25 @@ impl<T: BeaconChainTypes> SyncManager<T> {
             .filter(|req| req.state == BlockRequestsState::ReadyToProcess)
         {
             // verify the last added block is the parent of the last requested block
-            let last_index = completed_request.downloaded_blocks.len() - 1;
-            let expected_hash = completed_request.downloaded_blocks[last_index].parent_root;
-            // Note: the length must be greater than 1 so this cannot panic.
-            let block_hash = completed_request.downloaded_blocks[last_index - 1].canonical_root();
+
+            assert!(
```

Comment:
Can we add a `crit` for this too?

Reply:
This is the one we really need to worry about. I fully expect in a few runs to hit this and crash. I've not fully debugged it, but this will help and in a future iteration once I'm sure, can remove this assert.